intern/openmesheffect/blender/intern/mfxCallbacks.cpp

```diff
@@ -179,30 +387,80 @@ OfxStatus before_mesh_release(OfxHost *host, OfxMeshHandle ofx_mesh) {
+  ps->propGetInt(&ofx_mesh->properties, kOfxMeshPropNoLooseEdge, 0, &ofx_no_loose_edge);
+  ps->propGetInt(&ofx_mesh->properties, kOfxMeshPropConstantFaceCount, 0, &ofx_constant_face_count);
+
+  if (ofx_point_count < 0 ||
+      ofx_vertex_count < 0 ||
+      ofx_face_count < 0 ||
+      (ofx_no_loose_edge != 0 && ofx_no_loose_edge != 1) ||
+      (ofx_no_loose_edge == 1 && ofx_constant_face_count == 2 && ofx_face_count > 0) ||
+      (ofx_face_count > 0 && (ofx_constant_face_count < 2 && ofx_constant_face_count != -1))) {
+    printf("WARNING: Bad mesh property values\n");
+    return kOfxStatErrBadHandle;
+  }
 
+  // We need to handle stride here. While buffers from meshAlloc() are contiguous, these attributes
+  // may have been forwarded from input mesh, in which case they would be strided.
   OfxPropertySetHandle pos_attrib, vertpoint_attrib, facecounts_attrib;
   mes->meshGetAttribute(ofx_mesh, kOfxMeshAttribPoint, kOfxMeshAttribPointPosition, &pos_attrib);
   ps->propGetPointer(pos_attrib, kOfxMeshAttribPropData, 0, (void**)&point_data);
+  ps->propGetInt(pos_attrib, kOfxMeshAttribPropStride, 0, &point_stride);
   mes->meshGetAttribute(ofx_mesh, kOfxMeshAttribVertex, kOfxMeshAttribVertexPoint, &vertpoint_attrib);
   ps->propGetPointer(vertpoint_attrib, kOfxMeshAttribPropData, 0, (void**)&vertex_data);
+  ps->propGetInt(vertpoint_attrib, kOfxMeshAttribPropStride, 0, &vertex_stride);
   mes->meshGetAttribute(ofx_mesh, kOfxMeshAttribFace, kOfxMeshAttribFaceCounts, &facecounts_attrib);
   ps->propGetPointer(facecounts_attrib, kOfxMeshAttribPropData, 0, (void**)&face_data);
+  ps->propGetInt(facecounts_attrib, kOfxMeshAttribPropStride, 0, &face_stride);
 
   ps->propSetPointer(&ofx_mesh->properties, kOfxMeshPropInternalData, 0, NULL);
 
-  if (NULL == point_data || NULL == vertex_data || NULL == face_data) {
+  if ((NULL == point_data && ofx_point_count > 0) ||
+      (NULL == vertex_data && ofx_vertex_count > 0) ||
+      (NULL == face_data && ofx_face_count > 0 && -1 == ofx_constant_face_count)) {
     printf("WARNING: Null data pointers\n");
     return kOfxStatErrBadHandle;
   }
 
+  // Figure out geometry size on Blender side.
+  // Separate true faces (polys) and 2-vertex faces (loose edges), to get proper faces/edges in Blender.
+  // This requires reinterpretation of OFX face and vertex attributes, since we'll "forget"
+  // vertices associated with loose edges:
+  //
+  // OFX 2-vertex face (ie. edge) -> Blender edge (no loops)
+  // OFX n-vertex face (ie. poly) -> Blender poly and loops
+  if (1 == ofx_no_loose_edge) {
+    loose_edge_count = 0;
+    if (ofx_constant_face_count == -1) {
+      assert(check_no_loose_edges_in_ofx_mesh(ofx_face_count, face_data, face_stride));
+    }
+  } else if (2 == ofx_constant_face_count) {
+    loose_edge_count = ofx_face_count;
+  } else {
+    loose_edge_count = 0;
+    for (int i = 0; i < ofx_face_count; ++i) {
+      int vertex_count = *(int*)((char*)face_data + i*face_stride);
```

Comment:
Make me think this casting pattern deserves a little inline utility function cause it's used many times!